lib/world.rb

```diff
@@ -8,13 +8,28 @@ def initialize(cells)
 
   def evolve
     next_generation = []
-      @cells.each {|cell|
-      neighbours_count = count_live_neighbours(cell.position)
-        if ideal_number_of_neighbours(neighbours_count)
-          next_generation.push(cell)
-        end
-      }
-      @cells = next_generation
+    @cells.each do |cell|
+      if survive?(cell.position)
+        next_generation.push(cell)
+      end
+    end
+    positions_of_potential_resuscitating_cells.each do |position|
+      if resuscitate?(position)
+        new_cell = new_instance_of_cell(position)
+        next_generation.push(new_cell)
+      end
+    end
+    @cells = next_generation
+  end
+
+  def survive?(cell_position)
+    neighbours_count = count_live_neighbours(cell_position)
```

Comment:
The method name `count_live_neighbours` can be shortened to `neighbour_count`. You're only counting the live cells as neighbours anyway, aren't you?